xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -343,76 +369,60 @@ XWiki.index.trash.documents.confirmDelete = function(event, table, rowIdx) {
   }
 }
 
-/**
- * Event listener, called when clicking the "delete all" button.
- *
- * @param event the link activation event
- */
-XWiki.index.trash.documents.confirmDeleteAll = function(event) {
-  event.stop();
-  if (this.hasClassName('disabled')) {
-    // Do nothing if the button was already clicked and it's waiting for a response from the server.
-    return;
-  } else {
-    new XWiki.widgets.ConfirmedAjaxRequest(
-      /* Ajax request URL */
-      XWiki.currentDocument.getURL('delete'),
-      /* Ajax request parameters */
-      {
-        parameters : {
-          'confirm': 1,
-          'emptybin': true,
-          'form_token': "$!{services.csrf.getToken()}",
-          'ajax': 1
-        },
-        onCreate : function() {
-          // Disable the button, to avoid a cascade of clicks from impatient users
-          this.addClassName('disabled');
-        }.bind(this),
-        onSuccess : function() {
-          // Update the livetable
-          trashDocumentsLivetable.refresh();
-        }.bindAsEventListener(this)
-      },
-      /* Interaction parameters */
-      {
-         confirmationText: "$services.localization.render('core.recyclebin.completelyDeleteConfirm')",
-         progressMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteInProgress')",
-         successMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteDone')",
-         failureMessageText : "$services.localization.render('platform.index.trashDocumentsDeleteFailed')"
-      }
-      );
-  }
-}
-/*
- * The livetable is hidden when javascript is disabled, so that an alternative &lt;noscript&gt; table is displayed.
- * Re-display it when the document loads and Javascript is available.
- */
-document.observe('xwiki:dom:loaded', function() {
-  if ($('documentsTrash')) {
-    $('documentsTrash').up('.hidden').removeClassName('hidden');
-  }
+require(['jquery', 'xwiki-meta', 'xwiki-events-bridge'], function($, xm) {
+  //
+  // Delete All
+  //
+  var deleteDocumentConfirmModal = $('#deleteDocumentConfirmModal');
+  deleteDocumentConfirmModal.on('show.bs.modal').find('.btn-danger').click(function() {
```

Comment:
This looks strange:
* ``on()`` expects an event handler function
* you don't need to wait for the modal to be shown to register the click listener
* if you register the click listener each time the modal is shown then it will be called multiple times

Reply:
That looks indeed weird. I'll change that.